kolena/_experimental/dataset/_dataset.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import mimetypes
+from dataclasses import asdict
+from typing import Iterator
+from typing import Union
+
+import pandas as pd
+
+from kolena._api.v2.dataset import LoadDatapointsRequest
+from kolena._api.v2.dataset import Path
+from kolena._api.v2.dataset import RegisterRequest
+from kolena._utils import krequests_v2 as krequests
+from kolena._utils.batched_load import _BatchedLoader
+from kolena._utils.batched_load import init_upload
+from kolena._utils.batched_load import upload_data_frame
+from kolena._utils.consts import BatchSize
+from kolena._utils.state import API_V2
+from kolena.errors import InputValidationError
+from kolena.workflow._datatypes import _deserialize_dataobject
+from kolena.workflow._datatypes import _serialize_dataobject
+from kolena.workflow._datatypes import DATA_TYPE_FIELD
+from kolena.workflow._datatypes import TypedDataObject
+
+COL_DATAPOINT = "datapoint"
+TEST_SAMPLE_TYPE = "TEST_SAMPLE"
+FIELD_LOCATOR = "locator"
+FIELD_TEXT = "text"
+
+_DATAPOINT_TYPE_MAP = {
+    "image": f"{TEST_SAMPLE_TYPE}/IMAGE",
+    "application/pdf": f"{TEST_SAMPLE_TYPE}/DOCUMENT",
+    "text": f"{TEST_SAMPLE_TYPE}/DOCUMENT",
+    "video": f"{TEST_SAMPLE_TYPE}/VIDEO",
+}
+
+
+def _dataobject_type(obj: TypedDataObject) -> str:
+    obj_type = obj._data_type()
+    return f"{obj_type._data_category()}/{obj_type.value}"
+
+
+def _get_datapoint_type(mimetype_str: str) -> str:
+    main_type, sub_type = mimetype_str.split("/")
+    return _DATAPOINT_TYPE_MAP.get(mimetype_str, None) or _DATAPOINT_TYPE_MAP.get(main_type, None)
+
+
+def _infer_datatype_value(x: str) -> str:
+    mtype, _ = mimetypes.guess_type(x)
+    if mtype:
+        datatype = _get_datapoint_type(mtype)
+        if datatype is not None:
+            return datatype
+    elif x.endswith(".pcd"):
```

Comment:
Is this not covered by `mimetypes.guess_type`?

Reply:
It does not. For reference, [mimetype](https://docs.python.org/3/library/mimetypes.html#mimetypes.guess_type) package references [this list ](https://www.iana.org/assignments/media-types/media-types.xhtml)